Citizen - Happiness - /Bug fix #2799

bob0bob opened the pull request:
On the citizen windows, there is now a "next" and "previous" button on the gui. This will allow you to toggle between the main view and the happiness view.

The Happiness view allows you to view the different modifiers that affect that citizens happiness. Each value has a color to it.
"RED" means has a negative affect on their happiness
"YELLOW" means that it has no affect on their happiness
"GREEN" means it has a positive affect on their happiness

Also, BUG. The no fields modifier was not checking to make sure they are a "FARMER" and was applying it to every citizen. That is why most citizen couldn't get above a value of 6.0

This window shows all modifiers and some of the modifiers will not apply to that citizen. Those values will be always "YELLOW" meaning no affect.

Having everything "YELLOW" will get you a value of 8. to get higher you need some in green.

Closes #
Closes #
Closes #

# Changes proposed in this pull request:
-
-
-

Review please

bob0bob committed:
Citizen - Happiness - /Bug fix #2799

On the citizen windows, there is now a "next" and "previous" button on the gui. This will allow you to toggle between the main view and the happiness view.

The Happiness view allows you to view the different modifiers that affect that citizens happiness. Each value has a color to it.
"RED" means has a negative affect on their happiness
"YELLOW" means that it has no affect on their happiness
"GREEN" means it has a positive affect on their happiness

Also, BUG. The no fields modifier was not checking to make sure they are a "FARMER" and was applying it to every citizen. That is why most citizen couldn't get above a value of 6.0

This window shows all modifiers and some of the modifiers will not apply to that citizen. Those values will be always "YELLOW" meaning no affect.

Having everything "YELLOW" will get you a value of 8. to get higher you need some in green.

bob0bob committed:
code changes to comply with code style.

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 40183 |  1  |  1  |  0   |
| Errors |  597  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=16347&tab=Inspection&buildTypeId=bt88)

bob0bob committed:
Changes on a constant name that was wrong.

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/WindowConstants.java:
```diff
@@ -696,6 +696,16 @@
     public static final String LABEL_CONSTRUCTION_NAME =  "constructionName";
```
### Errors:
- Field has no usages.

### Warnings:
- Constant name LABEL_CONSTRUCTION_NAME doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/WindowConstants.java:
```diff
@@ -696,6 +696,16 @@
     public static final String LABEL_CONSTRUCTION_NAME =  "constructionName";
     public static final String LABEL_CONSTRUCTION_POS =  "constructionPos";
```
### Errors:
- Field has no usages.

### Warnings:
- Constant name GUI_BUTTON_RECALCULATE doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/WindowConstants.java:
```diff
@@ -981,6 +991,21 @@
+    /**
+     * ID for happiness modifier view
+     */
+    public static final String HAPPINESS_MODIFIER_PANE = "happinessModifierView";
```
### Errors:
- Field has no usages.

### Warnings:
- Constant name HAPPINESS_MODIFIER_PANE doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/WindowConstants.java:
```diff
@@ -981,6 +991,21 @@
      */
     public static final int LIFE_COUNT_DIVIDER = 30;
```
### Errors:
- Field has no usages.

### Warnings:
- Constant name LIFE_COUNT_DIVIDER doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/WindowConstants.java:
```diff
@@ -696,6 +696,16 @@
+    /**
+     * The fields used into the happiness
+     */
+    public static final String FOOD_MODIFIER    = "foodLevel";
```
### Errors:
- Field has no usages.

### Warnings:
- Constant name FOOD_MODIFIER doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/WindowConstants.java:
```diff
@@ -696,6 +696,16 @@
+     * The fields used into the happiness
+     */
+    public static final String FOOD_MODIFIER    = "foodLevel";
+    public static final String HOUSE_MODIFIER   = "houseLevel";
```
### Errors:
- Field has no usages.

### Warnings:
- Constant name HOUSE_MODIFIER doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/WindowConstants.java:
```diff
@@ -981,6 +991,21 @@
+    /**
+     * Button leading the player to the next page.
+     */
+    public static final String BUTTON_PREV_PAGE = "prevPage";
```
### Errors:
- Field has no usages.

### Warnings:
- Constant name BUTTON_PREV_PAGE doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/WindowConstants.java:
```diff
@@ -696,6 +696,16 @@
+     */
+    public static final String FOOD_MODIFIER    = "foodLevel";
+    public static final String HOUSE_MODIFIER   = "houseLevel";
+    public static final String DAMAGE_MODIFIER  = "damageLevel";
```
### Errors:
- Field has no usages.

### Warnings:
- Constant name DAMAGE_MODIFIER doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/WindowConstants.java:
```diff
@@ -696,6 +696,16 @@
+    public static final String FOOD_MODIFIER    = "foodLevel";
+    public static final String HOUSE_MODIFIER   = "houseLevel";
+    public static final String DAMAGE_MODIFIER  = "damageLevel";
+    public static final String JOB_MODIFIER     = "jobLevel";
```
### Errors:
- Field has no usages.

### Warnings:
- Constant name JOB_MODIFIER doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/WindowConstants.java:
```diff
@@ -696,6 +696,16 @@
+    public static final String HOUSE_MODIFIER   = "houseLevel";
+    public static final String DAMAGE_MODIFIER  = "damageLevel";
+    public static final String JOB_MODIFIER     = "jobLevel";
+    public static final String FIELDS_MODIFIER  = "farmsLevel";
```
### Errors:
- Field has no usages.

### Warnings:
- Constant name FIELDS_MODIFIER doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/WindowConstants.java:
```diff
@@ -696,6 +696,16 @@
+    public static final String DAMAGE_MODIFIER  = "damageLevel";
+    public static final String JOB_MODIFIER     = "jobLevel";
+    public static final String FIELDS_MODIFIER  = "farmsLevel";
+    public static final String TOOLS_MODIFIER   = "toolsLevel";
```
### Errors:
- Field has no usages.

### Warnings:
- Constant name TOOLS_MODIFIER doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/WindowConstants.java:
```diff
@@ -981,6 +991,21 @@
+    /**
+     * Button leading the player to the previous page.
+     */
+    public static final String BUTTON_NEXT_PAGE = "nextPage";
```
### Errors:
- Field has no usages.

### Warnings:
- Constant name BUTTON_NEXT_PAGE doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java:
```diff
@@ -75,6 +75,16 @@
+    /**
+     * Button leading to the previous page.
+     */
+    private Button buttonPrevPage;
```
### Warnings:
- Instance field name buttonPrevPage doesn't match regex 'm_[a-z][A-Za-z\d]*'

LDTTeam-Buildsystem replied:
### Warnings:
- Instance field name buttonPrevPage doesn't match regex 'm_[a-z][A-Za-z\d]*'

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java:
```diff
@@ -75,6 +75,16 @@
+    /**
+     * Button leading to the next page.
+     */
+    private Button buttonNextPage;
```
### Warnings:
- Instance field name buttonNextPage doesn't match regex 'm_[a-z][A-Za-z\d]*'

LDTTeam-Buildsystem replied:
### Warnings:
- Instance field name buttonNextPage doesn't match regex 'm_[a-z][A-Za-z\d]*'

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java:
```diff
@@ -58,7 +58,7 @@
      * Scrollinglist of the resources.
      */
-    private final ScrollingList   resourceList;
+    private ScrollingList   resourceList;
```
### Warnings:
- Instance field name resourceList doesn't match regex 'm_[a-z][A-Za-z\d]*'

LDTTeam-Buildsystem replied:
### Warnings:
- Instance field name resourceList doesn't match regex 'm_[a-z][A-Za-z\d]*'

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java:
```diff
@@ -75,6 +75,16 @@
      */
     private       int             lifeCount  = 0;
```
### Warnings:
- Instance field name lifeCount doesn't match regex 'm_[a-z][A-Za-z\d]*'

LDTTeam-Buildsystem replied:
### Warnings:
- Instance field name lifeCount doesn't match regex 'm_[a-z][A-Za-z\d]*'

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 40182 |  2  |  3  |  -1  |
| Errors |  597  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=16371&tab=Inspection&buildTypeId=bt88)

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 40182 |  2  |  3  |  -1  |
| Errors |  597  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=16371&tab=Inspection&buildTypeId=bt88)

bob0bob committed:
changing happiness view

Changed the modifiers being displayed.  If they have no affect, then they will not be displayed.   So this will end up variable length on how many items will be displayed.

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java:
```diff
@@ -275,6 +275,20 @@
     public static final String ON_STRING                                                           = "com.minecolonies.coremod.gui.townHall.on";
```
### Errors:
- Field has no usages.

### Warnings:
- Constant name ON_STRING doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java:
```diff
@@ -275,6 +275,20 @@
+    @NonNls
+    public static final String CMCG_HAPPINESS_FARMS                                                = "com.minecolonies.coremod.gui.happiness.farms";
+    @NonNls
+    public static final String CMCG_HAPPINESS_TOOLS                                                = "com.minecolonies.coremod.gui.happiness.tools";
```
### Errors:
- Field has no usages.

### Warnings:
- Constant name CMCG_HAPPINESS_TOOLS doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java:
```diff
@@ -275,6 +275,20 @@
     public static final String ON_STRING                                                           = "com.minecolonies.coremod.gui.townHall.on";
     @NonNls
     public static final String OFF_STRING                                                          = "com.minecolonies.coremod.gui.townHall.off";
```
### Errors:
- Field has no usages.

### Warnings:
- Constant name OFF_STRING doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java:
```diff
@@ -275,6 +275,20 @@
     @NonNls
     public static final String OFF_STRING                                                          = "com.minecolonies.coremod.gui.townHall.off";
+    @NonNls
+    public static final String CMCG_HAPPINESS_HAPPINESSMODIFIER                                    = "com.minecolonies.coremod.gui.happiness.happinessModifier";
```
### Errors:
- Field has no usages.

### Warnings:
- Constant name CMCG_HAPPINESS_HAPPINESSMODIFIER doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java:
```diff
@@ -275,6 +275,20 @@
+    @NonNls
+    public static final String CMCG_HAPPINESS_HAPPINESSMODIFIER                                    = "com.minecolonies.coremod.gui.happiness.happinessModifier";
+    @NonNls
+    public static final String CMCG_HAPPINESS_FOOD                                                 = "com.minecolonies.coremod.gui.happiness.food";
```
### Errors:
- Field has no usages.

### Warnings:
- Constant name CMCG_HAPPINESS_FOOD doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java:
```diff
@@ -275,6 +275,20 @@
+    @NonNls
+    public static final String CMCG_HAPPINESS_FOOD                                                 = "com.minecolonies.coremod.gui.happiness.food";
+    @NonNls
+    public static final String CMCG_HAPPINESS_DAMAGE                                               = "com.minecolonies.coremod.gui.happiness.damage";
```
### Errors:
- Field has no usages.

### Warnings:
- Constant name CMCG_HAPPINESS_DAMAGE doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java:
```diff
@@ -275,6 +275,20 @@
+    @NonNls
+    public static final String CMCG_HAPPINESS_DAMAGE                                               = "com.minecolonies.coremod.gui.happiness.damage";
+    @NonNls
+    public static final String CMCG_HAPPINESS_HOUSE                                                = "com.minecolonies.coremod.gui.happiness.house";
```
### Errors:
- Field has no usages.

### Warnings:
- Constant name CMCG_HAPPINESS_HOUSE doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java:
```diff
@@ -275,6 +275,20 @@
+    @NonNls
+    public static final String CMCG_HAPPINESS_HOUSE                                                = "com.minecolonies.coremod.gui.happiness.house";
+    @NonNls
+    public static final String CMCG_HAPPINESS_JOB                                                  = "com.minecolonies.coremod.gui.happiness.job";
```
### Errors:
- Field has no usages.

### Warnings:
- Constant name CMCG_HAPPINESS_JOB doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java:
```diff
@@ -275,6 +275,20 @@
+    @NonNls
+    public static final String CMCG_HAPPINESS_JOB                                                  = "com.minecolonies.coremod.gui.happiness.job";
+    @NonNls
+    public static final String CMCG_HAPPINESS_FARMS                                                = "com.minecolonies.coremod.gui.happiness.farms";
```
### Errors:
- Field has no usages.

### Warnings:
- Constant name CMCG_HAPPINESS_FARMS doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java:
```diff
@@ -44,7 +44,7 @@
 
 import static com.minecolonies.api.util.constant.Suppression.RAWTYPES;
```
### Errors:
- Unused import import static com.minecolonies.api.util.constant.Suppression.RAWTYPES;

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java:
```diff
@@ -44,7 +44,7 @@
 
 import static com.minecolonies.api.util.constant.Suppression.RAWTYPES;
 import static com.minecolonies.api.util.constant.WindowConstants.*;
```
### Errors:
- Unused import import static com.minecolonies.api.util.constant.WindowConstants.*;

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java:
```diff
@@ -44,7 +44,7 @@
 
 import static com.minecolonies.api.util.constant.Suppression.RAWTYPES;
 import static com.minecolonies.api.util.constant.WindowConstants.*;
-
```
### Errors:
- Unused import import static com.minecolonies.api.util.constant.TranslationConstants.*;

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 40199 | 18  |  1  | +17  |
| Errors |  597  |  0  |  0  |  +0  |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=16413&tab=Inspection&buildTypeId=bt88)

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 40199 | 18  |  1  | +17  |
| Errors |  597  |  0  |  0  |  +0  |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=16413&tab=Inspection&buildTypeId=bt88)

Raycoms committed:
fix pos

Raycoms committed:
Merge branch 'version/1.12' into feature/citizenHappiness
